libraries/abstractions/common_io/test/test_iot_sdio.c

```diff
@@ -0,0 +1,1073 @@
+/*-----------------------------------------------------------*/
+void prvSdioCallback( IotSdioSlotHandle_t const pxSdioHandle,
+                      IotSdioEventType_t eSdioEvent, void * pvUserContext );
+
+
+/*-----------------------------------------------------------*/
+/* Globals values which can be overwritten by the test
+ * framework invoking these tests */
+/*-----------------------------------------------------------*/
+uint8_t uctestIotSdioValidHostIdx = 0;                                  /**< A valid host index number. */
+uint8_t uctestIotSdioInvalidHostIdx = 2;                                /**< A invalid host index number. */
+uint8_t uctestIotSdioValidSlotIdx = 0;                                  /**< A valid Slot index number. */
+uint8_t uctestIotSdioInvalidSlotIdx = 1;                                /**< A invalid Slot index number. */
+uint8_t uctestIotSdioInvalidFuncNmbr = 9;                               /**< A invalid i/o function number. */
+IotSdioCallback_t xtestIotSdioTestCallback = prvSdioCallback;           /**< A sdio callback function. */
+IotSdioCardDetectParam_t xtestIotSdioCardDetectParam =
+                                   {eDetectCardByHostCD, false, false}; /**< A sdio card detection params. */
+IotSdioBusWidth_t etestIotSdioBusWidth = eDataBus4Bit;                  /**< sdio data bus width */
+IotSdioSdDriverStrength_t etestIotSdioDriverType =
+                                   eSdDriverStrengthTypeD;               /**< Driver strength type. */
+bool btestIotSdioVerifyInactive = true;                                  /**< Verify card inactive state or not. */
+
+
+/*-----------------------------------------------------------*/
+/** Static globals */
+/*-----------------------------------------------------------*/
+static const uint32_t g_funcTupleList[testIotSdio_FUNC_TPL_CNT] = {
+    testIotSdio_TPL_CODE_FUNCID, testIotSdio_TPL_CODE_FUNCE,
+};
+
+extern IotSdioSlotHandle_t gIotSdioHandle[ testIotSdio_HANDLE_HOST_NUM ][ testIotSdio_HANDLE_SLOT_NUM ];
+/*-----------------------------------------------------------*/
+
+/* Define Test Group. */
+TEST_GROUP( TEST_IOT_SDIO );
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Setup function called before each test in this group is executed.
+ */
+TEST_SETUP( TEST_IOT_SDIO )
+{
+    IotSdioSlotHandle_t xGlobalSdioHandle = gIotSdioHandle[ testIotSdio_HOST_INSTANCE ][ testIotSdio_SLOT_INSTANCE ];
+    int32_t lRetVal;
+
+    /* Close global handle it was opened. */
+    if( xGlobalSdioHandle != NULL )
+    {
+        lRetVal = iot_i2c_close( xGlobalSdioHandle );
+        TEST_ASSERT_EQUAL( IOT_SDIO_SUCCESS, lRetVal );
+
+        gIotSdioHandle[ testIotSdio_HOST_INSTANCE ][ testIotSdio_SLOT_INSTANCE ] = NULL;
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Tear down function called after each test in this group is executed.
+ */
+TEST_TEAR_DOWN( TEST_IOT_SDIO )
```

Comment:
Should the global handle be closed in the test tear down as well ?

Reply:
The global is closed (is open) in test setup so the tests can open SDIO to run test.  When the tests are done, there is no need to check it again.